src/charm.py

```diff
@@ -2458,9 +2458,16 @@ def relations_user_databases_map(self) -> dict:
             return {USER: "all", REPLICATION_USER: "all", REWIND_USER: "all"}
         user_database_map = {}
         try:
-            for user in self.postgresql.list_users_from_relation(
-                current_host=self.is_connectivity_enabled
-            ):
+            for user in self.postgresql.list_users(current_host=self.is_connectivity_enabled):
+                if user in (
+                    "backup",
+                    "monitoring",
+                    "operator",
+                    "postgres",
+                    "replication",
+                    "rewind",
+                ):
+                    continue
```

Comment:
Copied from the trigger.

Reply:
Please create a dedicated Jira ticket on Mohamed to decide the behavior here, as we are adding here backward compatibility to the current behavior which is not the best way to do in modernly hardened charm.

Thank you!